### ISSUE TITLE
CLI: accept templating for command line arguments

### ISSUE THREAD

**Issue by baby-gnu** [saltstack/salt#61605]
Hello.

**Is your feature request related to a problem? Please describe.**

When executing modules from command line, I often require looking up some grains or pillars before actually doing the work.

As an example, listing the `minion.d` configuration directory on all minions require knowing where to find the directory and generally execute command by targeting per `os_family`.

Recently, as windows `root_dir` and `config_dir` [changed in `3004`](https://docs.saltproject.io/en/latest/topics/installation/windows.html#installation), it became more complicated.

**Describe the solution you'd like**

I would like to be able to execute a command like:

```
salt '*' file.readdir '{{ salt["config.get"]("config_dir") | path_join("minion.d") }}' template=jinja
```

**Describe alternatives you've considered**

I have no clean way of doing things, I just target per `os_family` which was acceptable until the directory change on `3004` for windows.

**Additional context**

This is already done for [`cmd.run`](https://docs.saltproject.io/en/latest/ref/modules/all/salt.modules.cmdmod.html#salt.modules.cmdmod.run):

```
salt '*' cmd.run 'echo "{{ salt["config.get"]("config_dir") | path_join("minion.d") }}"' template=jinja
```

**Comment by waynew** [comment]
FWIW it looks like cmd.run specifically takes a `template` argument, which is rendered within the module itself.

I wonder if we could do something more generically like we do with requisites.

**Comment by waynew** [comment]
Had a chat with the core team on this and there's nothing necessarily _wrong_ with this idea, but there are a lot of things that would need to be in place first to make this not super buggy and error-prone.

Basically, we should do this but it will not happen any time soon. In the meantime, shell substitution is a good workaround (and a handy skill to have if you're maintaining servers _anyway_)

```
❯ salt-call file.readdir $(salt-call config.get config_dir --out=json | jq ".local" -r)/minion.d
local:
    - .
    - ..
    - grains.conf
    - _schedule.conf
```
    
Obviously it's not 1:1 (and states should be preferred *anyway*, which already offer this functionality), but hopefully it's enough for now.

**Comment by waynew** [comment]
Some of the challenges that would make this less-than-straightforward is that the command line would have to be rendered for every minion individually, before being sent to the states & modules. There are parts of that process that are... complicated. Some of the APIs involved in that process aren't well defined, or perhaps implemented consistently or completely across the backing services. So all of that would really need to be nailed down with robust tests before we would even consider implementing this feature.

**Comment by baby-gnu** [comment]
Thanks @waynew.

So for now, I will stay with my `/srv/salt/onetime/` sls tree :stuck_out_tongue_winking_eye:
